### DISCUSSION TITLE
question abt Scintillua features [too general and rather unrelated question, asked and answered in another discussion]

### DISCUSSION THREAD

**Discussion by dr-dba** [orbitalquark/scintillua#38] [Q&A] [answered]
Hello, a bit of background - i am struggling about idea to find an ideal editor. That is the general motivation.
I have relatively strong grip on Lua programming, gradually get into it thru programming API of FAR manager.
Lets call FAR manager (http://farmanager.com/) "far-mgr" for brevity.
So coded for it, trying to make of far-mgr built-in programmable editor something to go.
the far-mgr editor can be surprisingly very limited in some places, although really good at other
So, the experience not too much luck for me it,
beside learnining Lua eventually, and other kinds of "selfdevelopment" probably

Thus and herafter, i look now towards TextAdept adoption,
as for example, I need to know, how Scintillua can help starting with:

*   certain line hiding (folding/colapsing)
*   grepping abilities.

As I understand, folding text is related to lexing, which is related to ScintiLua,
which look attractive due my some knowledge about Lua programming.
Together with that, i would be more than happy to get some of advice,
and some general answers, before i am go into it too deep.. :)

Right now i know very little if at all of TextAdept and ScintiLua,
badly need some at least general advice.

I hope i explained myself, will be glad to continue any related discussion,
from my side - i am sharing all my coding, so that all the community can get advance,
nothing to boast about, i am just trying to explain myself, and my pursuits

**Comment by orbitalquark** [comment, marked as answer]
Scintillua provides syntax highlighting primarily for the Scintilla[1] editing component, though it has a Lua interface for use with other editors. While it can identify fold points in source code, Scintillua cannot fold arbitrary lines. There are no grepping abilities.

Basically, text editors have editing components, which may support syntax highlighting. Textadept is an editor based on the Scintilla editing component which supports syntax highlighting through Scintillua.

[1]: https://scintilla.org
